asf_public_discourse_home_decarbonisation/pipeline/faqs_identification/extract_questions.py

```diff
@@ -0,0 +1,445 @@
+
+    Args:
+        sentences (List[str]): A list of sentences to be evaluated. Each item in the list should be a string representing a single sentence.
+        inclusion_phrases (List[str]): A list of inclusion phrases. Each item in the list should be a string representing a phrase to search for in the sentences.
+
+    Returns:
+        idk_phrases (List[str]):
+            - This list contains sentences that include any of the predefined "don't know" phrases.
+
+    Note:
+        - The function is case-insensitive when searching for the predefined phrases.
+    """
+    idk_phrases = []
+    for sentence in sentences:
+        if any(phrase in sentence.lower() for phrase in inclusion_phrases):
+            idk_phrases.append(sentence)
+    return idk_phrases
+
+
+def extract_X_from_idk(
+    idk_phrases: List[str], inclusion_phrases: List[str]
+) -> List[str]:
+    """
+    Extracts and returns a list of sentences without the inclusion phrases found in the "I don't know" phrases.
+
+    The function takes a list of "I don't know" phrases and a list of inclusion phrases. It searches for sentences in the "I don't know" phrases that contain any of the inclusion phrases. It then extracts the text following the inclusion phrase and returns a list of these extracted sentences.
+
+    Args:
+        idk_phrases (List[str]): A list of "I don't know" phrases. Each item in the list should be a string representing a single sentence.
+        inclusion_phrases (List[str]): A list of inclusion phrases. Each item in the list should be a string representing a phrase to search for in the "I don't know" phrases.
+
+    Returns:
+        sentences_without_inclusion (List[str]):
+            - This list contains sentences extracted from the "I don't know" phrases, excluding the inclusion phrases.
+
+    Note:
+        - The function is case-insensitive when searching for the inclusion phrases.
+        - If an inclusion phrase is not found in a sentence, that sentence will not be included in the output list.
+    """
+    sentences_without_inclusion = []
+    for sentence in idk_phrases:
+        for phrase in inclusion_phrases:
+            if phrase.lower() in sentence.lower():
+                start_index = sentence.lower().index(phrase.lower()) + len(phrase)
+                extracted_text = sentence[start_index:].strip()
+                sentences_without_inclusion.append(extracted_text)
+                break
+    return sentences_without_inclusion
+
+
+def logger_statistics(category_dataframe):
+    """
+    Calculates and logs basic statistics about the category-specific data.
+
+    Args:
+        category_dataframe (DataFrame): The category-specific data.
+
+    Note:
+        - The function calculates the total number of "idk" phrases, questions, and sentences in the dataframe.
+        - The statistics are logged using the `logging.info()` method.
```

Comment:
Rather than adding a note at the end, you can make the docstring more explicit at the start as I do below. Also, it's unnecessary to mention that your using the logging.info() method as a note.

```suggestion
    Calculates and logs basic statistics about the category-specific data:
          - total number of sentences;
          - total number of phrases mentioning "do not know" or "don't know";
          - total number of questions.

    Args:
        category_dataframe (DataFrame): The category-specific data.
```